pkg/detector/detector.go

```diff
@@ -537,9 +537,20 @@ func (d *ResourceDetector) GetUnstructuredObject(objectKey keys.ClusterWideKey)
 
 	object, err := d.InformerManager.Lister(objectGVR).Get(objectKey.NamespaceKey())
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			klog.Errorf("Failed to get object(%s), error: %v", objectKey, err)
+		if apierrors.IsNotFound(err) {
+			// If the target object is not found in the informer cache,
+			// use the DynamicClient to get the target object again.
+			object, err := d.DynamicClient.Resource(objectGVR).Namespace(objectKey.Namespace).
+				Get(context.TODO(), objectKey.Name, metav1.GetOptions{})
+			if err == nil {
+				return object, nil
+			}
+			if apierrors.IsNotFound(err) {
```

Comment:
how about deleting this part and letting line 553 print the error?

Reply:
Update